docker/nginx-consul-template/nginx.conf.tmpl

```diff
@@ -0,0 +1,32 @@
+events {
+    worker_connections 4096;
+}
+
+http {
+{{- with $service_query := (printf "%s.%s" (env "DE_ENV") (env "NGINX_PROXY_SERVICE_NAME")) }}
+  {{- if service $service_query }}
+    upstream service {
+    {{- range service $service_query }}
+      server {{.Address}}:{{.Port}} max_fails=3 fail_timeout=60 weight=1 {{- if not (.Tags.Contains (key_or_default (printf "%s/%s/color" (env "DE_ENV") (env "NGINX_PROXY_SERVICE_NAME")) "green"))}} backup{{end}}; # {{.ID}} {{.Tags}}
```

Comment:
a) I'm a little ambivalent about this. Automatically detecting and choosing a color that has services (or choosing all servers by default) certainly makes it easier to achieve zero-downtime deployments in the event of a misconfiguration. On the other hand, it also makes it slightly more likely an older version of the service will continue to be used after a deployment. It would also add some complexity to the template. Another advantage of _not_ building this check in is that we're likely to be notified quickly if service calls start producing errors. As long as we take pains not to have services deployed on nights or weekends, that may be preferable to having the wrong version of a service in use.

b) This case would be a little more difficult to troubleshoot once it's happened because nginx will still be running with an outdated configuration and it may or may not produce errors depending on whether or not the previously configured color has servers. If we could configure the template to detect this and alert us if this happens then that might be helpful.